### PR TITLE
Support for ipv6 on cluster set command.

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: coord/src/rmr/endpoint.c [hunk 1]
```diff
@@ -21,15 +21,31 @@ int MREndpoint_Parse(const char *addr, MREndpoint *ep) {
...
     addr = at + 1;
+  }
+
+  if (addr[0] == '[') {
+      ++addr; // skip the ipv6 opener '['
   }
+
+  char *iter = strchr(addr, ':');
+  char *colon = NULL;
+  while (iter) {
```
Comment (oshadmi) [inline coord/src/rmr/endpoint.c:32]: Should we fail with more than 7 colons or less than 2?

File: coord/src/rmr/endpoint.c [hunk 2]
```diff
@@ -21,15 +21,31 @@ int MREndpoint_Parse(const char *addr, MREndpoint *ep) {
...
 
-  char *colon = strchr(addr, ':');
-  if (!colon || colon == addr) {
+  if (!colon) {
     MREndpoint_Free(ep);
     return REDIS_ERR;
+  }
+
+  size_t s = colon - addr;
+  if (addr[s - 1] == ']') {
```
Comment (oshadmi) [inline coord/src/rmr/endpoint.c:44]: Should we skip closing `]` only if found an opening `[`?

File: coord/src/rmr/test/test_cluster.c [hunk 3]
```diff
@@ -14,11 +14,26 @@
...
   mu_check(!strcmp(ep.host, "localhost"));
-  mu_assert_int_eq(6379, ep.port);
+  mu_assert_int_eq(6379, ep.port);
+
+  // ipv6 tests
+  mu_assert_int_eq(REDIS_OK, MREndpoint_Parse("::0:6379", &ep));
+  mu_check(!strcmp(ep.host, "::0"));
+  mu_assert_int_eq(6379, ep.port);
+
+  mu_assert_int_eq(REDIS_OK, MREndpoint_Parse("[fe80::8749:8fe8:f206:2ab9]:6379", &ep));
```
Comment (oshadmi) [inline coord/src/rmr/test/test_cluster.c:28]: Cover also brackets without port?

File: coord/src/rmr/test/test_cluster.c [hunk 4]
```diff
@@ -14,11 +14,26 @@
...
+  mu_check(!strcmp(ep.host, "fe80::8749:8fe8:f206:2ab9"));
+  mu_assert_int_eq(6379, ep.port);
+
+  mu_assert_int_eq(REDIS_OK, MREndpoint_Parse("pass@[fe80::8749:8fe8:f206:2ab9]:6379", &ep));
```
Comment (oshadmi) [inline coord/src/rmr/test/test_cluster.c:32]: Cover also password + brackets without port?

File: coord/src/rmr/test/test_cluster.c (hunk 3 above)
Comment (oshadmi) [reply coord/src/rmr/test/test_cluster.c:28]: Maybe use a different port than the default?

Comment (MeirShpilraien) [reply coord/src/rmr/test/test_cluster.c]: Added

File: coord/src/rmr/test/test_cluster.c (hunk 4 above)
Comment (MeirShpilraien) [reply coord/src/rmr/test/test_cluster.c]: Added

**Review by oshadmi — APPROVED** [review]

**Review by rafie — DISMISSED** [review]

**Review by rafie — APPROVED** [review]

**Pull request merged**